src/tests/Register.test.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+
+import ShallowRenderer from 'react-test-renderer/shallow';
+import { render, fireEvent, screen, waitForElement } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import Register from "../register/Register";
+
+
+test('renders Register', () => {
+  const renderer = new ShallowRenderer();
+  renderer.render(<Register />);
+  expect(renderer.getRenderOutput()).toMatchSnapshot()
+});
+
+// Mock successful registration
+const server = setupServer(
+  rest.post('/register', (req, res, ctx) => {
+    return res(ctx.json({ message: "User was created successfully. A confirmation email has been sent via email. After confirming your email you can login." }))
+  })
+)
+
+beforeAll(() => server.listen())
+afterEach(() => server.resetHandlers())
+afterAll(() => server.close())
+
+test('allows the user to register successfully', async () => {
+  render(<Register />)
+
+  fireEvent.change(screen.getByPlaceholderText('Full Name'), {
+    target: {value: 'My name'},
+  })
+  fireEvent.change(screen.getByPlaceholderText('Username'), {
+    target: {value: 'MyUsername'},
+  })
+  fireEvent.change(screen.getByPlaceholderText('Email'), {
+    target: {value: 'myemail@xxx.xxx'},
+  })
+  fireEvent.change(screen.getByPlaceholderText('Password'), {
+    target: {value: '12345678'},
+  })
+  fireEvent.change(screen.getByRole('checkbox', {name: "mentor"}), {
+    target: {value: 'true'},
+  })
+  fireEvent.change(screen.getByRole('checkbox', {name: "mentee"}), {
+    target: {value: 'true'},
+  })
+  fireEvent.change(screen.getByRole('checkbox', {name: "termsCheck"}), {
+    target: {value: 'true'},
+  })
+  fireEvent.change(screen.getByRole('button', {name: "Sign Up"}), {
```

Comment:
You can just click on the button. 
fireEvent.click(screen.getByText("Sign Up"));

Reply:
Yes, we can do either way. Just saw using getByRole is preferred way from [this](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library) article.

Reply:
Sure, thanks for sharing :) I was aiming at using fireEvent.click instead of change for a button